### ISSUE TITLE
VSA Design collaboration review for BAM2 [Financial Status Report (FSR)]

### ISSUE THREAD

**Issue by rileyorr** [department-of-veterans-affairs/va.gov-team#15867]
## Team Name

Benefits & Memorials 2

## Product/Feature Name

Financial Status Report (FSR)

## Product description

There are multiple debt resolutions available for Veterans that incur a VA debt. If the Veteran wants to request a waiver, compromise offer, or extended payment plan, they need to submit a Financial Status Report (FSR). The FSR's goal is to capture a monthly snapshot of the Veteran's financial situation along with information about why they're requesting a waiver, compromise offer, or extended payment plan. 

## Review stage

- [X] **Design collaboration review** (Review design to discuss accessibility and inclusive design) - This may happen as design evolves.

- [ ] **Development intent review** (Review design to discuss approach to take with development, identify what components and semantic HTML would be appropriate, identify any accessibility issues and how to address) - This would happen as designs are passed to development, before development begins.

- [ ] **Spot check review** (Review and follow the steps in the [508 Checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/508-checklist.md) first.) This happens well before launch, so that the 508 office can review and any issues addressed *before* launch.

- [ ] **Other:** Please indicate what you're requesting

## Point of contact

Riley Orr
Rebecca Walsh
Leah Keeler 

## Preferred times for dev intent checkpoint, if applicable

N/A - would like asynchronous feedback, but I am open to meeting if it would be helpful!

## Staging URL and credentials, if applicable

Our [in-progress design](https://preview.uxpin.com/803dc24213cf723bfce6c5e3eaef5a2c3f3e344b#/pages/133701377/simulate/sitemap?mode=i) currently lives in UXPin. 

I'm particularly looking for feedback on our list loop designs. We have a couple variations going on which incorporate the 526-EZ's list loop, Ryan Thurwell's new design, and the disability rating calculator. I'd love some feedback on accessibility considerations to make as we move toward code. 

**Note:** This is our first pass at hi-fi designs, and we still have a few chapters to design. I'll likely request another spot-check once those chapters are completed.

**Comment by jenstrickland** [comment]
Some of this feedback is likely better-suited for a "dev intent review" and I'm providing it here so that the devs have it sooner than later. 🤞 

This first batch of feedback is not around the "list loop pattern" implementation. I'll add that in the next comment.

1. I imagine that y'all already know this, and I'm being pedantic 🧟‍♀️ … by marking up the heading order.

![Screen Shot 2020-11-06 at 9 21 36 AM](https://user-images.githubusercontent.com/57469/98411513-2bd24e80-2044-11eb-8704-8b100821492d.png)

<hr/>

2. The "Sign in or create an account" button needs an `aria-describedby="uniqe_id"` to associate it with the following `h2` that introduces the supporting content. 
    - For screen reader users, they may hear button "Sign in or create an account"  and not know there is additional info. The `aria-describedby` will let them know there is supporting content.
    - The Healthcare team is considering proposing a change to this page, where this button would be at the bottom of the page, based on research that revealed users didn't read the content beyond the button, they just clicked clicked the button. It's very helpful info. 

![Screen Shot 2020-11-06 at 3 28 26 PM](https://user-images.githubusercontent.com/57469/98411987-1d386700-2045-11eb-9bb3-4a985f1ce925.png)

<hr/>

3. In the Need Help? section, the information structure, and especially the TTY number, formatting could be enhanced. Here's an example of the information currently presented on VA.gov.

![Screen Shot 2020-11-06 at 3 40 45 PM](https://user-images.githubusercontent.com/57469/98412784-8c628b00-2046-11eb-904b-5d7e12b2e1ae.png)

<hr/>

4. The steps have been updated to remove the blue circle with white numbers. See the screenshot attached below for current styles.

### Current multi-step form styling
<img width="820" alt="Screen Shot 2020-10-09 at 12 01 29 PM" src="https://user-images.githubusercontent.com/934879/95611389-65a13c80-0a27-11eb-8dae-9f032985be51.png">

<hr/>

5. For the email input, there must be a pair of inputs so that a user confirms their email entry in order to meet WCAG 2.0 Level AA:
> [3.3.4 Error Prevention (Legal, Financial, Data)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-reversible.html): For Web pages that cause legal commitments or financial transactions for the user to occur, that modify or delete user-controllable data in data storage systems, or that submit user test responses, at least one of the following is true: 
> • Reversible: Submissions are reversible.
> • Checked: Data entered by the user is checked for input errors and the user is provided an opportunity to correct them.
> • Confirmed: A mechanism is available for reviewing, confirming, and correcting information before finalizing the submission.

![Screen Shot 2020-11-06 at 4 10 22 PM](https://user-images.githubusercontent.com/57469/98415647-aa7eba00-204b-11eb-9841-5da03df06b98.png)

**Comment by rileyorr** [comment]
Hi, @jenstrickland! Thank you again for providing feedback and collaborating with me and @rebeccawalsh on a list loop design that'll work for our form! We've incorporated minor adjustments on the original design you and Josh proposed; I'd love a quick accessibility spot-check to see if there are any glaring issues. It's essentially the same design, but we'd like to display the saved inputs as a table instead of inside the card. We think it'll be easier for our users to scan the information while using the existing list loop functionality.

I've added our concept to a quick [UXPin prototype](https://preview.uxpin.com/fa323c51a84318fe2c5aec9cf6f125fa4062625b#/pages/134282568/simulate/no-panels?mode=i) for you to take a look at! Please let me know if you have any questions/feedback. Thank you!

**Comment by jenstrickland** [comment]
Thanks, @rileyorr & @rebeccawalsh for the quick turnaround. I provided comments in the UXPin prototype. Please let me know if you want to hop on a call, if there are any questions I may answer async, or if y'all want to schedule a collab session with Ryan if the mobile layout suggests that design system default mobile styling changes should be recommended. I think the mobile view could be for an iPhone 6 to show a realistic use case.

**Comment by rebeccawalsh** [comment]
Thanks, @jenstrickland.

For the remove functionality during edit mode, are you imagining the white/blue-border buttons aligned right? (Just based on the pin point of your comment) 
![image](https://user-images.githubusercontent.com/6361930/99578781-52c94280-29ab-11eb-82fc-38c96a665447.png)

As for mobile styling, we're leaning most toward stacking the content, since it's very possible this won't always be 2-field pairings, and we're supposed to accommodate 280px screens.

**Comment by jenstrickland** [comment]
@rebeccawalsh That's what I suspect is right. Whatever it was that @rtwell  had in his card design that has the Save, Cancel, and Remove functionality.

**Comment by rileyorr** [comment]
@jenstrickland I forgot to mention this, but we'd like to incorporate 'search suggestion' (unsure if this is the correct terminology) functionality that exists in the 526-EZ's list loop component. This would allow users to see predetermined values  as well as include additional values we didn't provide. Are there any accessibility considerations we should take into account when it comes to that functionality? I've attached a screenshot of the component I'm referencing from the 526-EZ.

<img width="606" alt="Screen Shot 2020-11-18 at 4 56 12 PM" src="https://user-images.githubusercontent.com/46329147/99593294-71d1cf80-29bf-11eb-9a73-9e19e7a3f011.png">

**Comment by jenstrickland** [comment]
@rileyorr Yes, this component exists already in a few places, referred to as "type ahead." The Public Websites team recently used a `<datalist>` for the search functionality on their CMS pages. Facility Locator also has a type ahead. 

> … should use the Autocomplete component from the VSP design team. However, this component has known 508 compliance issues, which should be addressed, perhaps by the VSP team. Screen readers will speak every character typed, then the full word, which may collide with the speaking of the search results.
> Expected behavior: Screen-reader users are notified when the list of suggestions appears and that they can enter the list and select an option using the keyboard.

Recommendations are to wait until 3 characters have been entered before proposing suggestions to reduce the noise for screen reader users. When the word is announced after each character, screen readers begin to speak the word suggestions and it sounds disorienting. I recommend connecting with @ncksllvn about the datalist his team implemented. From the testing, it seemed that was useful. 

* Adam Silver has a great resource on delivering an accessible autocomplete control: https://adamsilver.io/articles/building-an-accessible-autocomplete-control/

* [MDN datalist element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist)

* [HTML5 datalist, what you need to know](https://www.jotform.com/blog/html5-datalists-what-you-need-to-know-78024/)

* [Jeremy Keith has done quite a bit with the datalist](https://adactio.com/journal/17337), and ensuring that it has solid fallbacks for browsers that do not support it. Check out his implementation on [The Session](https://thesession.org/).

**Comment by jenstrickland** [comment]
@rileyorr 👋  As your team moves to develop the typeahead functionality, this [recent spot check review for the VSA search team ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16965)may be of interest. Their implementation had some unique constraints that I don't think are relevant for your team.

**Comment by ncksllvn** [comment]
There is a VSA Search & Discovery team now that does the typeahead work. They're on the `#va-search` channel on Slack. I don't think they use a `datalist` anymore but I'm not certain so don't quote me on that.

**Comment by jenstrickland** [comment]
@ncksllvn That's the ticket I just referenced above. It's the VSA search team and a spot check review they had. I figured providing the documentation itself was helpful.

**Comment by ncksllvn** [comment]
Oh. I just saw my username mentioned in the email notification (old comment it looks like now but I just saw it) and want to make sure folks go to the right place for typeahead questions (which isn't me)

**Comment by srsuddath** [comment]
Hey all, I stumbled upon this conversation while clicking through ticket references. I am the front-end engineer working on Search and Discovery. If you have any questions about typeahead experiences and why we chose Downshift vs Datalist, please feel free to reach out in a mention or on our ```#va-search``` slack channel
